main.wasm.go

```diff
@@ -0,0 +1,211 @@
+	"fmt"
+	"io"
+	"syscall/js"
+
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"strconv"
+
+	"github.com/HexmosTech/httpie-go/exchange"
+	"github.com/HexmosTech/httpie-go/flags"
+	"github.com/HexmosTech/httpie-go/input"
+	"github.com/HexmosTech/httpie-go/output"
+	"github.com/pkg/errors"
+)
+
+type Options struct {
+	// Transport is applied to the underlying HTTP client. Use to mock or
+	// intercept network traffic.  If nil, http.DefaultTransport will be cloned.
+	Transport http.RoundTripper
+}
+
+type ExResponse struct {
+	StatusCode int
+	Body       string
+	Headers    map[string]string
+}
+
+func getStringSlice(jsArray js.Value) []string {
+	length := jsArray.Length()
+	slice := make([]string, length)
+	for i := 0; i < length; i++ {
+		slice[i] = jsArray.Index(i).String()
+	}
+	return slice
+}
+
+func Int(i int) int {
+	i, _ = strconv.Atoi(strconv.Itoa(i))
+	return i
+}
+
+func Lama2Entry(cmdArgs []string, stdinBody io.Reader, proxyURL string, proxyUsername string, proxyPassword string, autoRedirect bool) (ExResponse, error) {
+	options := Options{}
+	args, usage, optionSet, err := flags.Parse(cmdArgs)
+	if err != nil {
+		fmt.Println(err)
+		return ExResponse{}, err
+	}
+	inputOptions := optionSet.InputOptions
+	exchangeOptions := optionSet.ExchangeOptions
+	exchangeOptions.Transport = options.Transport
+	outputOptions := optionSet.OutputOptions
+	in, err := input.ParseArgs(args, stdinBody, &inputOptions)
+	if _, ok := errors.Cause(err).(*input.UsageError); ok {
+		usage.PrintUsage(os.Stderr)
+		return ExResponse{}, err
+	}
+	if err != nil {
+		fmt.Println(err)
+		return ExResponse{}, err
+	}
+
+	// Send request and receive response
+	status, err := Exchange(in, &exchangeOptions, &outputOptions, proxyURL, proxyUsername, proxyPassword, autoRedirect)
+	if err != nil {
+		fmt.Println(err)
+		return ExResponse{}, err
+	}
+
+	if exchangeOptions.CheckStatus {
+		os.Exit(getExitStatus(status.StatusCode))
+	}
+
+	return status, nil
+}
+
+func getExitStatus(statusCode int) int {
+	if 300 <= statusCode && statusCode < 600 {
+		return statusCode / 100
+	}
+	return 0
+}
+
+func Exchange(in *input.Input, exchangeOptions *exchange.Options, outputOptions *output.Options, proxyURL string, proxyUsername string, proxyPassword string, autoRedirect bool) (ExResponse, error) {
+	// Prepare printer
+	writer := bufio.NewWriter(os.Stdout)
+	defer writer.Flush()
+	printer := output.NewPrinter(writer, outputOptions)
+	// Build HTTP request
+	request, err := exchange.BuildHTTPRequest(in, exchangeOptions)
+	if err != nil {
+		fmt.Println(err)
+		return ExResponse{-1, "", map[string]string{}}, err
+	}
+	username := "proxyServer"
+	password := "proxy22523146server"
+	auth := fmt.Sprintf("%s:%s", username, password)
+	basic := "Basic " + base64.StdEncoding.EncodeToString([]byte(auth))
+	request.Header.Add("proxyauth", basic)
+	cookieValue := request.Header.Get("Cookie")
+	if cookieValue != "" {
+		request.Header.Add("CustomCookie", cookieValue)
+	}
+	request.RequestURI = ""
```

Comment:
What does this line do?

Reply:
We need to set the URL and clear the RequestURI if you want to send the request using a client.  [ more here](https://stackoverflow.com/questions/19595860/http-request-requesturi-field-when-making-request-in-go#:~:text=The%20error%20is%20clear%3A%20You,when%20doing%20a%20client%20request.&text=to%20a%20server.,in%20an%20HTTP%20client%20request.)